src/compiler/crystal/tools/playground/server.cr

```diff
@@ -471,14 +472,19 @@ module Crystal::Playground
         end
       end
 
+
+      address = @host
+      address = "localhost" if address == "127.0.0.1" || address == "localhost" || address == nil
+
+
       client_ws = PathWebSocketHandler.new "/client" do |ws, context|
         origin = context.request.headers["Origin"]
         if !accept_request?(origin)
           Log.warn { "Invalid Request Origin: #{origin}" }
           ws.close :policy_violation, "Invalid Request Origin"
         else
           @sessions_key += 1
-          @sessions[@sessions_key] = session = Session.new(ws, @sessions_key, @port)
+          @sessions[@sessions_key] = session = Session.new(ws, @sessions_key, @port, address.to_s)
```

Comment:
If change to ``` @sessions[@sessions_key] = session = Session.new(ws, @sessions_key, address)``` will get error ```Error: undefined local variable or method 'address' for Crystal::Playground::Server```, So to fix this will need define a new class variable ```@address``` and modify the arguments api of function ```initialize``` and ```instrument_and_prelude``` of Class ````Session```,     I didn't do this only for not modify too much code~

I am newbie for Crystal and PR, and I am very happy and appreciate for you give time to guide me to finish this PR, thanks a lot~

Reply:
Sure! You'll need to replace the `@port` instance variable with `@address`. And all uses of that instance variable should use `address.port` instead because the port is included in the address.

Reply:
If  still need me to go on modify it, just call me~ Thanks~